### ISSUE TITLE
Rule proposal: `v-for-delimiter-style`

### ISSUE THREAD

**Issue by FloEdelmann** [vuejs/eslint-plugin-vue#1228]
**Please describe what the rule should do:**
You can use both `v-for="x in xs"` and `v-for="x of xs"`. There should be a rule to enforce one of the styles (probably configurable).

**What category should the rule belong to?**

- [X] Enforces code style
- [ ] Warns about a potential error
- [ ] Suggests an alternate way of doing something
- [ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**
```vue
<!-- "vue/v-for-delimiter-style": ["warn", "in"] -->
<div v-for="x in xs" /><!-- GOOD -->
<div v-for="x of xs" /><!-- BAD -->

<!-- "vue/v-for-delimiter-style": ["warn", "of"] -->
<div v-for="x of xs" /><!-- GOOD -->
<div v-for="x in xs" /><!-- BAD -->
```

**Additional context**
See https://vuejs.org/v2/guide/list.html#Mapping-an-Array-to-Elements-with-v-for:

> You can also use `of` as the delimiter instead of `in`, so that it is closer to JavaScript’s syntax for iterators.

**Comment by ota-meshi** [comment]
Thank you for your proposal!

Sounds good to me.

However, looking at [the rules of eslint](https://eslint.org/docs/rules/), it seems that the meaning is a little different from the rules that have `consistent` as a prefix.
I think names like `vue/v-for-delimiter` or `vue/v-for-delimiter-style` are good. What do you think?

**Comment by FloEdelmann** [comment]
Indeed. I'd go with `vue/v-for-delimiter-style` then.

**Comment by ota-meshi** [comment]
If the PR to the parser is unacceptable, you can get it from the token. (it's difficult to read...)

```js
const tokenStore =
    context.parserServices.getTemplateBodyTokenStore &&
    context.parserServices.getTemplateBodyTokenStore()

const token = /** @type {Token} */ (tokenStore.getTokenAfter(
  node.left.length ? node.left[node.left.length-1] : tokenStore.getFirstToken(node),
  token => token.type !== 'Punctuator' || token.value !== ')'
))

token.value // "in" or "of"
```

The following sources are helpful.

https://github.com/vuejs/eslint-plugin-vue/blob/a2614ef22d33f98af2d9b300bd88e86b8a35c02f/lib/utils/indent-common.js#L1218
